src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -227,12 +227,78 @@ public ISqlConnectionFactory ConnectionFactory
         // Attempts to link a URI to an actively used connection for this URI
         public virtual bool TryFindConnection(string ownerUri, out ConnectionInfo connectionInfo) => this.OwnerToConnectionMap.TryGetValue(ownerUri, out connectionInfo);
 
+        /// <summary>
+        /// Refreshes the auth token of a given connection 
+        /// </summary>
+        /// <param name="ownerUri">The URI of the connection</param>
+        /// <returns>true upon successful refresh of the auth token,
+        /// false if the uri cannot be found</returns>

```

Comment:
the description doesn't match the behavior. false can also mean no refresh needed

Reply:
and if refresh fails.